VM Transform not working from a provider page

lgalis opened the pull request:
Move the vm_transform action to a mixin 

Links 
-------
https://bugzilla.redhat.com/show_bug.cgi?id=1514032


Steps for Testing/QA
-------------------------------
Steps to Reproduce:
1. Browse CFME UI -> Compute -> Infrastructure -> Providers
2. Select VMware 
3. click on VMs
4. Select lifecycle -> Transform tagged VMs to RHV.

With this PR, the dialog opens from ems_infra:

![screenshot from 2017-11-17 01-54-53](https://user-images.githubusercontent.com/12769982/32934672-942fd432-cb3a-11e7-9d0f-3fc5ffe08e57.png)

lgalis commented:
/cc @martinpovolny

lgalis commented:
@miq-bot assign @martinpovolny

martinpovolny commented on app/controllers/vm_infra_controller.rb:
```diff
@@ -2,6 +2,7 @@ class VmInfraController < ApplicationController
   include VmCommon # common methods for vm controllers
   include VmRemote # methods for VM remote access
   include VmShowMixin
+  include Mixins::Actions::VmActions::Transform
```
@lgalis , is this line needed? I'd say that the mixing is already included in ApplicationController throught CiProcessing and this controller inherits from Application controller.

lgalis commented:
@miq-bot add_label bug, gaprindashvili/yes

lgalis commented:
@martinpovolny - addressed your comment. There is one more thing that would need to be addressed - when the dialog is launched from the provider vm list page - the return from the dialog does not include the display paramter. I will include that in a follow up PR as it is a general dialog runner issue, not vm_transform specifc.

martinpovolny commented:
Seems the new specs are failing.

lgalis commented:
@martinpovolny - I pushed a change to the spec y mistake - reverted that

simaishi commented:
Gaprindashvili backport details:

```
$ git log -1
commit a55b882f66b5ce7ef949c7bc51fb046ad6016656
Author: Martin Povolny <mpovolny@redhat.com>
Date:   Sat Nov 18 08:48:57 2017 +0100

    Merge pull request #2757 from lgalis/fix_vm_transform_from_infra_provider
    
    VM Transform not working from a provider page
    (cherry picked from commit 80fb23f834f6cec1ff99d8ae6f84b7e91151ae03)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1515303
```
